d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancer.java

```diff
@@ -53,13 +56,12 @@
  * In DUAL_READ mode, it reads from both the old and the new load balancer, but relies on the data from old
  * load balancer only.
  */
-public class DualReadLoadBalancer implements LoadBalancerWithFacilities
-{
+public class DualReadLoadBalancer implements LoadBalancerWithFacilities {
   private static final Logger LOG = LoggerFactory.getLogger(DualReadLoadBalancer.class);
   private final LoadBalancerWithFacilities _oldLb;
   private final LoadBalancerWithFacilities _newLb;
   private final DualReadStateManager _dualReadStateManager;
-
+  private ThreadPoolExecutor _loadBalancerThreadPool;
```

Comment:
Nit: the name of the class is `DualReadLoadBalancer`, so it's redundant to call this field `_loadBalancer...`. The type of the field will change to `ExecutorService` according to my other comments, so it's best to just call it `_executor`

Reply:
Make sense. Rename the field to `executor`